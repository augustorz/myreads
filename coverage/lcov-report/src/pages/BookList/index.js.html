<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/pages/BookList/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/pages/BookList</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Link } from 'react-router-dom';
import { instanceOf, shape, func } from 'prop-types';
import { SHELVES } from './constants';
import './index.css';
&nbsp;
import BookShelf from '../../components/BookShelf';
&nbsp;
const BookList = ({ shelves, onChangeBookShelf }) =&gt; (
  &lt;div className="list-books"&gt;
    &lt;div className="list-books-title"&gt;
      &lt;h1&gt;MyReads&lt;/h1&gt;
    &lt;/div&gt;
    &lt;div className="list-books-content"&gt;
      {SHELVES.map(({ title, key }) =&gt; (
        &lt;BookShelf
          key={key}
          title={title}
          books={shelves[key]}
          onChangeBookShelf={onChangeBookShelf}
        /&gt;
      ))}
    &lt;/div&gt;
    &lt;div className="open-search"&gt;
      &lt;Link to="/search"&gt;
        &lt;button type="button"&gt;Add a book&lt;/button&gt;
      &lt;/Link&gt;
    &lt;/div&gt;
  &lt;/div&gt;
);
&nbsp;
BookList.propTypes = {
  shelves: shape({
    currentlyReading: instanceOf(Array),
    wantToRead: instanceOf(Array),
    read: instanceOf(Array),
  }).isRequired,
  onChangeBookShelf: func.isRequired,
};
&nbsp;
export default BookList;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Apr 01 2019 20:39:37 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
