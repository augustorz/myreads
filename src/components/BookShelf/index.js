import React from 'react';
import PropTypes from 'prop-types';

import Book from '../Book';

const BookShelf = ({ title, books, onChangeBookShelf }) => (
  <div className="bookshelf">
    <h2 className="bookshelf-title">{title}</h2>
    <div className="bookshelf-books">
      <ol className="books-grid">
        {books.map(book => (
          <Book key={book.id} book={book} onChangeBookShelf={onChangeBookShelf} />
        ))}
      </ol>
    </div>
  </div>
);

BookShelf.propTypes = {
  title: PropTypes.string.isRequired,
  books: PropTypes.instanceOf(Array),
  onChangeBookShelf: PropTypes.func.isRequired,
};

BookShelf.defaultProps = {
  books: [],
};

export default BookShelf;
