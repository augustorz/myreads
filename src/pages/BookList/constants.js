export const SHELVES = [
  { title: 'Current Reading', key: 'currentlyReading' },
  { title: 'Want to Read', key: 'wantToRead' },
  { title: 'Read', key: 'read' },
];

export default SHELVES;
