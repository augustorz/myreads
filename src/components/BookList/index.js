import React, { Component } from 'react';
import * as BooksAPI from '../../BooksAPI';

import BookShelf from '../BookShelf'

const SHELVES = [
  { title: 'Current Reading', key: 'currentlyReading' },
  { title: 'Want to Read', key: 'wantToRead' },
  { title: 'Read', key: 'read' }
]

class BookList extends Component {
  state = {
    shelves: {}
  }

  componentDidMount = () => {
    BooksAPI.getAll().then((books) => {
      this.addBooksToShelves(books)
    }, (error) => {
      console.log(error)
    })
  }

  removeBookFromShelf = ({shelves, book}) => {
    return shelves[book.shelf].filter(({ id }) => (
      id !== book.id 
    ))
  }

  addBookToShelf = ({shelves, book, newShelf}) => {
    book.shelf = newShelf

    if (newShelf !== 'none') {
      shelves[newShelf].push(book)
    }
  }

  changeBookShelf = ({book, newShelf}) => {
    let { shelves } = this.state
    
    shelves[book.shelf] = this.removeBookFromShelf({shelves, book})
    this.addBookToShelf({shelves, book, newShelf})
    this.setState({ shelves })
  }

  addBooksToShelves = (books = []) => {
    let shelves = {};

    books.forEach((book) => {
      if (!shelves[book.shelf]) {
        shelves[book.shelf] = [];
      }

      shelves[book.shelf].push(book);
    })
    
    this.setState({ shelves });
  }

  render () {
    const { shelves } = this.state;

    return (
      <div className="list-books">
        <div className="list-books-title">
          <h1>MyReads</h1>
        </div>
        <div className="list-books-content">
          {SHELVES.map(({ title, key }) => (
            <BookShelf key={key} title={title} books={shelves[key]} onChangeBookShelf={this.changeBookShelf} />
          ))}
        </div>
        <div className="open-search">
          <button onClick={() => this.setState({ showSearchPage: true })}>Add a book</button>
        </div>
      </div>
    )
  }
}

export default BookList;