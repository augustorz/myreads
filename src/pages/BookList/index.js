import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { SHELVES } from './constants';
import './index.css';
import * as BooksAPI from '../../api/BooksAPI';

import BookShelf from '../../components/BookShelf';

class BookList extends Component {
  componentDidMount = async () => {
    const { onMount } = this.props;

    onMount();
    this.loadBooks();
  }

  loadBooks = async () => {
    const { onBooksLoad } = this.props;

    try {
      const books = await BooksAPI.getAll();
      onBooksLoad(books);
    } catch (error) {
      throw (error);
    }
  }

  render() {
    const { shelves, onChangeBookShelf } = this.props;

    return (
      <div className="list-books">
        <div className="list-books-title">
          <h1>MyReads</h1>
        </div>
        <div className="list-books-content">
          {SHELVES.map(({ title, key }) => (
            <BookShelf
              key={key}
              title={title}
              books={shelves[key]}
              onChangeBookShelf={onChangeBookShelf}
            />
          ))}
        </div>
        <div className="open-search">
          <Link to="/search">
            <button type="button">Add a book</button>
          </Link>
        </div>
      </div>
    );
  }
}

BookList.propTypes = {
  shelves: PropTypes.shape({
    currentlyReading: PropTypes.instanceOf(Array),
    wantToRead: PropTypes.instanceOf(Array),
    read: PropTypes.instanceOf(Array),
  }).isRequired,
  onChangeBookShelf: PropTypes.func.isRequired,
  onBooksLoad: PropTypes.func.isRequired,
  onMount: PropTypes.func.isRequired,
};

export default BookList;
