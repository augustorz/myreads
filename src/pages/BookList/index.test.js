import React from 'react';
import { mount } from 'enzyme';
import { waitForElement } from 'enzyme-async-helpers';
import { MemoryRouter } from 'react-router-dom';
import createBooks from '../../mocks/books';

import BookList from '.';
import BookShelf from '../../components/BookShelf';

describe('BookList', () => {
  let wrapper;

  beforeEach(() => {
    fetch.resetMocks();
  });

  describe('componentDidMount', () => {
    beforeEach(async (done) => {
      const currentlyReading = createBooks({ quantity: 2, shelf: 'currentlyReading' });
      const wantToRead = createBooks({ quantity: 3, shelf: 'wantToRead' });
      const read = createBooks({ quantity: 1, shelf: 'read' });

      fetch.mockResponse(JSON.stringify({
        books: [
          ...currentlyReading,
          ...wantToRead,
          ...read,
        ],
      }));

      wrapper = mount(
        <MemoryRouter>
          <BookList
            shelves={{}}
            onBooksLoad={jest.fn()}
            onChangeBookShelf={jest.fn()}
          />
        </MemoryRouter>,
      );

      await waitForElement(wrapper, BookShelf);
      done();
    });

    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
