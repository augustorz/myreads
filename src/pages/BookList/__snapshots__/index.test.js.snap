// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList should match snapshot 1`] = `"<div class=\\"list-books\\"><div class=\\"list-books-title\\"><h1>MyReads</h1></div><div class=\\"list-books-content\\"><div class=\\"bookshelf\\"><h2 class=\\"bookshelf-title\\">Current Reading</h2><div class=\\"bookshelf-books\\"><ol class=\\"books-grid\\"><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width: 128px; height: 193px; background-image: url(thumbnail_0.png);\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 0</div><div class=\\"book-authors\\">Title Mock 0</div></div></li><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width: 128px; height: 193px; background-image: url(thumbnail_1.png);\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 1</div><div class=\\"book-authors\\">Title Mock 1</div></div></li></ol></div></div><div class=\\"bookshelf\\"><h2 class=\\"bookshelf-title\\">Want to Read</h2><div class=\\"bookshelf-books\\"><ol class=\\"books-grid\\"><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width: 128px; height: 193px; background-image: url(thumbnail_0.png);\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 0</div><div class=\\"book-authors\\">Title Mock 0</div></div></li><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width: 128px; height: 193px; background-image: url(thumbnail_1.png);\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 1</div><div class=\\"book-authors\\">Title Mock 1</div></div></li><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width: 128px; height: 193px; background-image: url(thumbnail_2.png);\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 2</div><div class=\\"book-authors\\">Title Mock 2</div></div></li></ol></div></div><div class=\\"bookshelf\\"><h2 class=\\"bookshelf-title\\">Read</h2><div class=\\"bookshelf-books\\"><ol class=\\"books-grid\\"><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width: 128px; height: 193px; background-image: url(thumbnail_0.png);\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 0</div><div class=\\"book-authors\\">Title Mock 0</div></div></li></ol></div></div></div><div class=\\"open-search\\"><a href=\\"/search\\"><button type=\\"button\\">Add a book</button></a></div></div>"`;
